File: unity/oelMVCS/Assets/XTC/oelMVCS/Scripts/Framework.cs
using System.Collections;
using System.Collections.Generic;

/*
         ______________________________
        \|/      \|/                   |
    Service -> Model -> Controller -> View
        /|\      /|\         |
         |________|__________|
*/

namespace XTC.oelMVCS
{
    public class Framework
    {
        public StaticPipe getStaticPipe()
        {
            return staticPipe_;
        }

        public DynamicPipe getDynamicPipe()
        {
            return dynamicPipe_;
        }

        public void setConfig(Config _value)
        {
            board_.setConfig(_value);
        }

        public void setLogger(Logger _value)
        {
            board_.setLogger(_value);
        }

        public Framework()
        {
            board_ = new Board();
            staticPipe_ = new StaticPipe(board_);
            dynamicPipe_ = new DynamicPipe(board_);
        }

        ~Framework()
        {
            board_ = null;
            staticPipe_ = null;
            dynamicPipe_ = null;
        }

        /// <summary>
        /// 框架初始化，完成各层中心的实例化
        /// </summary>
        public void Initialize()
        {
            board_.getLogger().Info("initialize framework");
            board_.setViewCenter(new ViewCenter(board_));
            board_.setModelCenter(new ModelCenter(board_));
            board_.setControllerCenter(new ControllerCenter(board_));
            board_.setServiceCenter(new ServiceCenter(board_));
        }

        /// <summary>
        /// 框架安装，完成各层中心已注册组件的安装
        /// 此过程将调用各派生组件的setup方法
        /// </summary>
        public void Setup()
        {
            board_.getLogger().Info("setup framework");
            board_.getViewCenter().Setup();
            board_.getServiceCenter().Setup();
            board_.getModelCenter().Setup();
            board_.getControllerCenter().Setup();
        }

        /// <summary>
        /// 框架拆卸，完成各层中心已注册组件的拆卸
        /// 此过程将调用各派生组件的dismantle方法
        /// </summary>
        public void Dismantle()
        {
            board_.getLogger().Info("dismantle framework");
            board_.getViewCenter().Dismantle();
            board_.getServiceCenter().Dismantle();
            board_.getModelCenter().Dismantle();
            board_.getControllerCenter().Dismantle();
        }


        /// <summary>
        /// 框架销毁，完成各层中心的释放
        /// </summary>
        public void Release()
        {
            board_.getLogger().Info("release framework");
            board_.setViewCenter(null);
            board_.setServiceCenter(null);
            board_.setModelCenter(null);
            board_.setControllerCenter(null);
        }

        private StaticPipe staticPipe_ = null;
        private DynamicPipe dynamicPipe_ = null;
        public Config config_ = null;
        public Logger logger_ = null;
        private Board board_ = null;
    }

}//namespace

File: python/oelMVCS/Framework.py
class Framework:
    def __init__(self):
        self.__board: Board = Board()
        self.__staticPipe : StaticPipc = StaticPipe(self.__board)
        self.__dynamicPipe : DynamicPipc = DynamicPipe(self.__board)

    def getStaticPipe(self):
        return self.staticPipe_

    def getDynamicPipe(self):
        return self.dynamicPipe_

    def setConfig(self, _config : Config):
        self.__board.setConfig(_config);

    def setLogger(self, _logger: Logger):
        self.__board.setLogger(_logger);


    def void Initialize(self):
        '''
        框架初始化，完成各层中心的实例化
        '''
        board_.logger.Info("initialize framework");
        board_.viewCenter = new ViewCenter(board_);
        board_.modelCenter = new ModelCenter(board_);
        board_.controllerCenter = new ControllerCenter(board_);
        board_.serviceCenter = new ServiceCenter(board_);

    def Setup(self):
        '''
        框架安装，完成各层中心已注册组件的安装
        此过程将调用各派生组件的setup方法
        '''
        board_.logger.Info("setup framework");
        board_.viewCenter.Setup();
        board_.serviceCenter.Setup();
        board_.modelCenter.Setup();
        board_.controllerCenter.Setup();

    def Dismantle(self):
        '''
        框架拆卸，完成各层中心已注册组件的拆卸
        此过程将调用各派生组件的dismantle方法
        '''
        board_.logger.Info("dismantle framework");
        board_.viewCenter.Dismantle();
        board_.serviceCenter.Dismantle();
        board_.modelCenter.Dismantle();
        board_.controllerCenter.Dismantle();


    def Release(self):
        '''
        框架销毁，完成各层中心的释放
        '''
        board_.logger.Info("release framework");
        board_.viewCenter = null;
        board_.serviceCenter = null;
        board_.modelCenter = null;
        board_.controllerCenter = null;
